### PR TITLE
add label selector to clusterset spec

### PR DESCRIPTION
**Pull request by ldpliu**
Signed-off-by: ldpliu <daliu@redhat.com>

### CODE REVIEW

File: cluster/v1beta1/types_managedclusterset.go [hunk 1]
```diff
@@ -47,18 +47,23 @@ type ManagedClusterSetSpec struct {
 
 // ManagedClusterSelector represents a selector of ManagedClusters
 type ManagedClusterSelector struct {
-	// SelectorType could only be "LegacyClusterSetLabel" now, will support more SelectorType later
+	// SelectorType could only be "LegacyClusterSetLabel" or "LabelSelector"
 	// "LegacyClusterSetLabel" means to use label "cluster.open-cluster-management.io/clusterset:<ManagedClusterSet Name>"" to select target clusters.
+	// "LabelSelector" means use labelSelector to select target managedClusters
 	// +kubebuilder:validation:Enum=LegacyClusterSetLabel
```
Comment (qiujian16) [inline cluster/v1beta1/types_managedclusterset.go]: need to update this

File: cluster/v1beta1/types_managedclusterset.go [hunk 2]
```diff
@@ -47,18 +47,23 @@ type ManagedClusterSetSpec struct {
...
 	// +kubebuilder:default:=LegacyClusterSetLabel
 	// +required
 	SelectorType SelectorType `json:"selectorType,omitempty"`
+
+	// LabelSelector define the general labelSelector which clusterset will use to select target managedClusters
```
Comment (qiujian16) [inline cluster/v1beta1/types_managedclusterset.go:58]: add optional and omitempty

File: cluster/v1beta1/types_managedclusterset.go [hunk 3]
```diff
@@ -47,18 +47,23 @@ type ManagedClusterSetSpec struct {
...
+	LabelSelector *metav1.LabelSelector `json:"labelSelector"`
 }
 
 type SelectorType string
 
 const (
 	LegacyClusterSetLabel SelectorType = "LegacyClusterSetLabel"
+	LabelSelector         SelectorType = "LabelSelector"
```
Comment (qiujian16) [inline cluster/v1beta1/types_managedclusterset.go]: add doc for each type

**Review by qiujian16 — COMMENTED** [review]
please add a test on this

File: cluster/v1beta1/types_managedclusterset.go (hunk 3 above)
Comment (ldpliu) [reply cluster/v1beta1/types_managedclusterset.go]: done

File: cluster/v1beta1/types_managedclusterset.go (hunk 2 above)
Comment (ldpliu) [reply cluster/v1beta1/types_managedclusterset.go:58]: done

File: cluster/v1beta1/types_managedclusterset.go (hunk 1 above)
Comment (ldpliu) [reply cluster/v1beta1/types_managedclusterset.go]: done

**Review by qiujian16 — COMMENTED** [review]
/approve

**Pull request merged**